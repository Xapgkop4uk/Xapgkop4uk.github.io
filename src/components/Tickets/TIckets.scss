.tickets-wrapper {
  display: flex;
  flex-direction: column;
  padding: 0 1rem;
  .ticket-wrapper {
    display: flex;
    width: 566px;
    height: 161px;
    background: #ffffff;
    margin-bottom: 1rem;
    box-shadow: 0px 1px 4px 0px rgba(91, 137,164, 0.25);
    .ticket-price {
      display: flex;
      flex-direction: column;
      padding: 1rem 0;
      justify-content: space-around;
      align-items: center;
      width: 200px;
      border-right: 1px solid rgba(0,0,0,0.1);
      .buy-ticket-button {
        width: 160px;
        height: 56px;
        background: #ff6d00;
        border: none;
        border-radius: 0.3rem;
        box-shadow: 0px 2px 1px 0px rgba(0,0,0,0.1);
        box-shadow: 0px 1px 0px 0px #d64d08;
        cursor: pointer;
        .buy-ticket-label {
          color: white;
          font-size: 16px;
          word-wrap: break-word;

        }
      }
    }
    .ticket-details {
      width: 366px;
      display: flex;
      .from {
        width: 25%;
        display: flex;
        justify-content: space-around;
        align-items: flex-start;
        flex-direction: column;
        padding: 1rem;
        color: #4a4a4a;
        .departure-time {
          font-size: 32px;
        }
        .departure-details {
          padding: 0 0.4rem 3.5rem;
          .port {
            font-size: 12px;
            white-space: nowrap;
          }
          .departure-date {
            color: #888;
            display: flex;
            font-size: 11px;
            justify-content: center;
            flex-direction: column;
            align-items: flex-start;
          }
        }
      }
      .to {
        width: 25%;
        display: flex;
        justify-content: space-around;
        align-items: flex-end;
        flex-direction: column;
        padding: 1rem;
        color: #4a4a4a;
        .arrival-time {
          font-size: 32px;
        }
        .arrival-details {
          padding: 0 0.4rem 3.5rem;
          .port {
            font-size: 12px;
            white-space: nowrap;
          }
          .arrival-date {
            color: #888;
            display: flex;
            font-size: 11px;
            justify-content: center;
            flex-direction: column;
            align-items: flex-end;
          }
        }
      }
      .route-line {
        width: 50%;
        position: relative;
        margin-bottom: 7.6rem;
        border-bottom: 1px solid rgba(155, 155, 155, 0.5);
        .route {
          text-transform: uppercase;
          color: black;
          font-size: 10px;
        }
        .plane {
          position: absolute;
          top: 2rem;
          right: -15px;
        }
      }
    }
  }
}