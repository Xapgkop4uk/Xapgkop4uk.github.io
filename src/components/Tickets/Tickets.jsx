import React from 'react'
import propTypes from 'prop-types'
import moment from 'moment'

const getTransfers = (count) => {
  switch (count) {
    case 0:
      return ''
    case 1:
      return '1 пересадка'
    case count > 4:
      return `${count} пересадок`
    default:
      return `${count} пересадки`
  }
}

const getPrice = (rates, selectedCurrency, price) => {
  if (!rates[selectedCurrency]) return price
  return rates[selectedCurrency] * price
}


const Tickets = ({ tickets, rates, selectedCurrency }) => (
  <div className="tickets-wrapper">
    {tickets.map(ticket => (
      <div className="ticket-wrapper">
        <div className="ticket-price">
          <img src="https://a.icons8.com/aUSOiWgn/DW0mD2/logo.svg" alt="avia-logo" />
          <button type="button" className="buy-ticket-button" onClick={() => {}}>
            <span className="buy-ticket-label">
              Купить
              <br />
              {`за ${new Intl.NumberFormat('ru-RU', { style: 'currency', currency: selectedCurrency, minimumFractionDigits: 0 }).format(getPrice(rates, selectedCurrency, ticket.price))}`}
            </span>
          </button>
        </div>
        <div className="ticket-details">
          <div className="from">
            <span className="departure-time">
              {ticket.departure_time}
            </span>
            <div className="departure-details">
              <span className="port">
                {`${ticket.origin}, ${ticket.origin_name}`}
              </span>
              <span className="departure-date">
                {moment(ticket.departure_date).format('DD MMM YYYY, ddd')}
              </span>
            </div>
          </div>
          <div className="route-line">
            <span className="route">
              {getTransfers(ticket.stops)}
            </span>
            <img src="https://a.icons8.com/hnhcupih/CIFYqd/ic_airplanemode_active_bl.svg" alt="plane" className="plane" />
          </div>
          <div className="to">
            <span className="arrival-time">
              {ticket.arrival_time}
            </span>
            <div className="arrival-details">
              <span className="port">
                {`${ticket.destination}, ${ticket.destination_name}`}
              </span>
              <span className="arrival-date">
                {moment(ticket.arrival_date).format('DD MMM YYYY, ddd')}
              </span>
            </div>
          </div>
        </div>
      </div>
    ))}
  </div>
)

Tickets.propTypes = {
  tickets:          propTypes.array.isRequired,
  rates:            propTypes.array.isRequired,
  selectedCurrency: propTypes.string.isRequired
}

export default Tickets
