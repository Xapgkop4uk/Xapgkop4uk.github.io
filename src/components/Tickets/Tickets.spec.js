import React from 'react'
import renderer from 'react-test-renderer'
import Tickets from './Tickets'
import source from '../../tickets.json'

const { tickets } = source

describe('Tickets container', () => {
  let el

  beforeAll(() => {
    el = renderer.create(<Tickets
      tickets={tickets}
      rates={{ EUR: 0.0133977497, USD: 0.0152479789 }}
      selectedCurrency="RUB"
    />)
  })
  it('just render without error', () => {
    expect(el).toBeDefined()
  })
  it('contains tickets', () => {
    const tree = el.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
