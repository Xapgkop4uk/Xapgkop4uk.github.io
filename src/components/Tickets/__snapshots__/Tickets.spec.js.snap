// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tickets container contains tickets 1`] = `
<div
  className="tickets-wrapper"
>
  <div
    className="ticket-wrapper"
  >
    <div
      className="ticket-price"
    >
      <img
        alt="avia-logo"
        src="https://a.icons8.com/aUSOiWgn/DW0mD2/logo.svg"
      />
      <button
        className="buy-ticket-button"
        onClick={[Function]}
        type="button"
      >
        <span
          className="buy-ticket-label"
        >
          Купить
          <br />
          за RUB 12,400
        </span>
      </button>
    </div>
    <div
      className="ticket-details"
    >
      <div
        className="from"
      >
        <span
          className="departure-time"
        >
          16:20
        </span>
        <div
          className="departure-details"
        >
          <span
            className="port"
          >
            VVO, Владивосток
          </span>
          <span
            className="departure-date"
          >
            05 Dec 2018, Wed
          </span>
        </div>
      </div>
      <div
        className="route-line"
      >
        <span
          className="route"
        >
          3 пересадки
        </span>
        <img
          alt="plane"
          className="plane"
          src="https://a.icons8.com/hnhcupih/CIFYqd/ic_airplanemode_active_bl.svg"
        />
      </div>
      <div
        className="to"
      >
        <span
          className="arrival-time"
        >
          22:10
        </span>
        <div
          className="arrival-details"
        >
          <span
            className="port"
          >
            TLV, Тель-Авив
          </span>
          <span
            className="arrival-date"
          >
            05 Dec 2018, Wed
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ticket-wrapper"
  >
    <div
      className="ticket-price"
    >
      <img
        alt="avia-logo"
        src="https://a.icons8.com/aUSOiWgn/DW0mD2/logo.svg"
      />
      <button
        className="buy-ticket-button"
        onClick={[Function]}
        type="button"
      >
        <span
          className="buy-ticket-label"
        >
          Купить
          <br />
          за RUB 13,100
        </span>
      </button>
    </div>
    <div
      className="ticket-details"
    >
      <div
        className="from"
      >
        <span
          className="departure-time"
        >
          17:20
        </span>
        <div
          className="departure-details"
        >
          <span
            className="port"
          >
            VVO, Владивосток
          </span>
          <span
            className="departure-date"
          >
            05 Dec 2018, Wed
          </span>
        </div>
      </div>
      <div
        className="route-line"
      >
        <span
          className="route"
        >
          1 пересадка
        </span>
        <img
          alt="plane"
          className="plane"
          src="https://a.icons8.com/hnhcupih/CIFYqd/ic_airplanemode_active_bl.svg"
        />
      </div>
      <div
        className="to"
      >
        <span
          className="arrival-time"
        >
          23:50
        </span>
        <div
          className="arrival-details"
        >
          <span
            className="port"
          >
            TLV, Тель-Авив
          </span>
          <span
            className="arrival-date"
          >
            05 Dec 2018, Wed
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ticket-wrapper"
  >
    <div
      className="ticket-price"
    >
      <img
        alt="avia-logo"
        src="https://a.icons8.com/aUSOiWgn/DW0mD2/logo.svg"
      />
      <button
        className="buy-ticket-button"
        onClick={[Function]}
        type="button"
      >
        <span
          className="buy-ticket-label"
        >
          Купить
          <br />
          за RUB 15,300
        </span>
      </button>
    </div>
    <div
      className="ticket-details"
    >
      <div
        className="from"
      >
        <span
          className="departure-time"
        >
          12:10
        </span>
        <div
          className="departure-details"
        >
          <span
            className="port"
          >
            VVO, Владивосток
          </span>
          <span
            className="departure-date"
          >
            05 Dec 2018, Wed
          </span>
        </div>
      </div>
      <div
        className="route-line"
      >
        <span
          className="route"
        >
          
        </span>
        <img
          alt="plane"
          className="plane"
          src="https://a.icons8.com/hnhcupih/CIFYqd/ic_airplanemode_active_bl.svg"
        />
      </div>
      <div
        className="to"
      >
        <span
          className="arrival-time"
        >
          18:10
        </span>
        <div
          className="arrival-details"
        >
          <span
            className="port"
          >
            TLV, Тель-Авив
          </span>
          <span
            className="arrival-date"
          >
            05 Dec 2018, Wed
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ticket-wrapper"
  >
    <div
      className="ticket-price"
    >
      <img
        alt="avia-logo"
        src="https://a.icons8.com/aUSOiWgn/DW0mD2/logo.svg"
      />
      <button
        className="buy-ticket-button"
        onClick={[Function]}
        type="button"
      >
        <span
          className="buy-ticket-label"
        >
          Купить
          <br />
          за RUB 11,000
        </span>
      </button>
    </div>
    <div
      className="ticket-details"
    >
      <div
        className="from"
      >
        <span
          className="departure-time"
        >
          17:00
        </span>
        <div
          className="departure-details"
        >
          <span
            className="port"
          >
            VVO, Владивосток
          </span>
          <span
            className="departure-date"
          >
            05 Dec 2018, Wed
          </span>
        </div>
      </div>
      <div
        className="route-line"
      >
        <span
          className="route"
        >
          2 пересадки
        </span>
        <img
          alt="plane"
          className="plane"
          src="https://a.icons8.com/hnhcupih/CIFYqd/ic_airplanemode_active_bl.svg"
        />
      </div>
      <div
        className="to"
      >
        <span
          className="arrival-time"
        >
          23:30
        </span>
        <div
          className="arrival-details"
        >
          <span
            className="port"
          >
            TLV, Тель-Авив
          </span>
          <span
            className="arrival-date"
          >
            05 Dec 2018, Wed
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ticket-wrapper"
  >
    <div
      className="ticket-price"
    >
      <img
        alt="avia-logo"
        src="https://a.icons8.com/aUSOiWgn/DW0mD2/logo.svg"
      />
      <button
        className="buy-ticket-button"
        onClick={[Function]}
        type="button"
      >
        <span
          className="buy-ticket-label"
        >
          Купить
          <br />
          за RUB 13,400
        </span>
      </button>
    </div>
    <div
      className="ticket-details"
    >
      <div
        className="from"
      >
        <span
          className="departure-time"
        >
          12:10
        </span>
        <div
          className="departure-details"
        >
          <span
            className="port"
          >
            VVO, Владивосток
          </span>
          <span
            className="departure-date"
          >
            05 Dec 2018, Wed
          </span>
        </div>
      </div>
      <div
        className="route-line"
      >
        <span
          className="route"
        >
          3 пересадки
        </span>
        <img
          alt="plane"
          className="plane"
          src="https://a.icons8.com/hnhcupih/CIFYqd/ic_airplanemode_active_bl.svg"
        />
      </div>
      <div
        className="to"
      >
        <span
          className="arrival-time"
        >
          20:15
        </span>
        <div
          className="arrival-details"
        >
          <span
            className="port"
          >
            TLV, Тель-Авив
          </span>
          <span
            className="arrival-date"
          >
            05 Dec 2018, Wed
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ticket-wrapper"
  >
    <div
      className="ticket-price"
    >
      <img
        alt="avia-logo"
        src="https://a.icons8.com/aUSOiWgn/DW0mD2/logo.svg"
      />
      <button
        className="buy-ticket-button"
        onClick={[Function]}
        type="button"
      >
        <span
          className="buy-ticket-label"
        >
          Купить
          <br />
          за RUB 12,450
        </span>
      </button>
    </div>
    <div
      className="ticket-details"
    >
      <div
        className="from"
      >
        <span
          className="departure-time"
        >
          9:40
        </span>
        <div
          className="departure-details"
        >
          <span
            className="port"
          >
            VVO, Владивосток
          </span>
          <span
            className="departure-date"
          >
            05 Dec 2018, Wed
          </span>
        </div>
      </div>
      <div
        className="route-line"
      >
        <span
          className="route"
        >
          3 пересадки
        </span>
        <img
          alt="plane"
          className="plane"
          src="https://a.icons8.com/hnhcupih/CIFYqd/ic_airplanemode_active_bl.svg"
        />
      </div>
      <div
        className="to"
      >
        <span
          className="arrival-time"
        >
          19:25
        </span>
        <div
          className="arrival-details"
        >
          <span
            className="port"
          >
            TLV, Тель-Авив
          </span>
          <span
            className="arrival-date"
          >
            05 Dec 2018, Wed
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ticket-wrapper"
  >
    <div
      className="ticket-price"
    >
      <img
        alt="avia-logo"
        src="https://a.icons8.com/aUSOiWgn/DW0mD2/logo.svg"
      />
      <button
        className="buy-ticket-button"
        onClick={[Function]}
        type="button"
      >
        <span
          className="buy-ticket-label"
        >
          Купить
          <br />
          за RUB 13,600
        </span>
      </button>
    </div>
    <div
      className="ticket-details"
    >
      <div
        className="from"
      >
        <span
          className="departure-time"
        >
          17:10
        </span>
        <div
          className="departure-details"
        >
          <span
            className="port"
          >
            VVO, Владивосток
          </span>
          <span
            className="departure-date"
          >
            05 Dec 2018, Wed
          </span>
        </div>
      </div>
      <div
        className="route-line"
      >
        <span
          className="route"
        >
          1 пересадка
        </span>
        <img
          alt="plane"
          className="plane"
          src="https://a.icons8.com/hnhcupih/CIFYqd/ic_airplanemode_active_bl.svg"
        />
      </div>
      <div
        className="to"
      >
        <span
          className="arrival-time"
        >
          23:45
        </span>
        <div
          className="arrival-details"
        >
          <span
            className="port"
          >
            TLV, Тель-Авив
          </span>
          <span
            className="arrival-date"
          >
            05 Dec 2018, Wed
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ticket-wrapper"
  >
    <div
      className="ticket-price"
    >
      <img
        alt="avia-logo"
        src="https://a.icons8.com/aUSOiWgn/DW0mD2/logo.svg"
      />
      <button
        className="buy-ticket-button"
        onClick={[Function]}
        type="button"
      >
        <span
          className="buy-ticket-label"
        >
          Купить
          <br />
          за RUB 14,250
        </span>
      </button>
    </div>
    <div
      className="ticket-details"
    >
      <div
        className="from"
      >
        <span
          className="departure-time"
        >
          6:10
        </span>
        <div
          className="departure-details"
        >
          <span
            className="port"
          >
            VVO, Владивосток
          </span>
          <span
            className="departure-date"
          >
            05 Dec 2018, Wed
          </span>
        </div>
      </div>
      <div
        className="route-line"
      >
        <span
          className="route"
        >
          
        </span>
        <img
          alt="plane"
          className="plane"
          src="https://a.icons8.com/hnhcupih/CIFYqd/ic_airplanemode_active_bl.svg"
        />
      </div>
      <div
        className="to"
      >
        <span
          className="arrival-time"
        >
          15:25
        </span>
        <div
          className="arrival-details"
        >
          <span
            className="port"
          >
            TLV, Тель-Авив
          </span>
          <span
            className="arrival-date"
          >
            05 Dec 2018, Wed
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ticket-wrapper"
  >
    <div
      className="ticket-price"
    >
      <img
        alt="avia-logo"
        src="https://a.icons8.com/aUSOiWgn/DW0mD2/logo.svg"
      />
      <button
        className="buy-ticket-button"
        onClick={[Function]}
        type="button"
      >
        <span
          className="buy-ticket-label"
        >
          Купить
          <br />
          за RUB 16,700
        </span>
      </button>
    </div>
    <div
      className="ticket-details"
    >
      <div
        className="from"
      >
        <span
          className="departure-time"
        >
          16:50
        </span>
        <div
          className="departure-details"
        >
          <span
            className="port"
          >
            VVO, Владивосток
          </span>
          <span
            className="departure-date"
          >
            05 Dec 2018, Wed
          </span>
        </div>
      </div>
      <div
        className="route-line"
      >
        <span
          className="route"
        >
          1 пересадка
        </span>
        <img
          alt="plane"
          className="plane"
          src="https://a.icons8.com/hnhcupih/CIFYqd/ic_airplanemode_active_bl.svg"
        />
      </div>
      <div
        className="to"
      >
        <span
          className="arrival-time"
        >
          23:35
        </span>
        <div
          className="arrival-details"
        >
          <span
            className="port"
          >
            TLV, Тель-Авив
          </span>
          <span
            className="arrival-date"
          >
            05 Dec 2018, Wed
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ticket-wrapper"
  >
    <div
      className="ticket-price"
    >
      <img
        alt="avia-logo"
        src="https://a.icons8.com/aUSOiWgn/DW0mD2/logo.svg"
      />
      <button
        className="buy-ticket-button"
        onClick={[Function]}
        type="button"
      >
        <span
          className="buy-ticket-label"
        >
          Купить
          <br />
          за RUB 17,400
        </span>
      </button>
    </div>
    <div
      className="ticket-details"
    >
      <div
        className="from"
      >
        <span
          className="departure-time"
        >
          6:10
        </span>
        <div
          className="departure-details"
        >
          <span
            className="port"
          >
            VVO, Владивосток
          </span>
          <span
            className="departure-date"
          >
            05 Dec 2018, Wed
          </span>
        </div>
      </div>
      <div
        className="route-line"
      >
        <span
          className="route"
        >
          
        </span>
        <img
          alt="plane"
          className="plane"
          src="https://a.icons8.com/hnhcupih/CIFYqd/ic_airplanemode_active_bl.svg"
        />
      </div>
      <div
        className="to"
      >
        <span
          className="arrival-time"
        >
          16:15
        </span>
        <div
          className="arrival-details"
        >
          <span
            className="port"
          >
            TLV, Тель-Авив
          </span>
          <span
            className="arrival-date"
          >
            05 Dec 2018, Wed
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
